### PR TITLE
docs: Add Mix Install/Docs, Start/Stop ABCI Server guides

### PR DESCRIPTION
**Pull request by ltfschoen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,10 +2,87 @@
...
+  ```
+
+* Run IEx
+  ```bash
+  iex -S mix
+  ```
+
+* Create a Fake Module named Foo
+  ```
+  iex(1)> defmodule Foo do
```
Comment (KrzysiekJ) [inline README.md:40]: This module doesn’t implement the `abci_app` behaviour, so it won’t be able to handle ABCI requests.

File: README.md [hunk 2]
```diff
@@ -2,10 +2,87 @@
...
+  ...(1)>   def bar() do
+  ...(1)>     IO.puts("You're using ABCI Server!")
+  ...(1)>   end
+  ...(1)> end
+  {:module, Foo,
+   <<70, ..., 117, ...>>, {:bar, 0}}
+  ```
+
+* Show ABCI Server Information (using `module_info/1` which is the Erlang equivalent of Elixir's `__info__/1`), Start ABCI Server, Stop ABCI Server
```
Comment (KrzysiekJ) [inline README.md:49]: This is a generic Erlang function. I don’t see much utility in putting information about this in README.

File: README.md [hunk 3]
```diff
@@ -2,10 +2,87 @@
...
+    ],
+    compile: [
+      ...
+    ],
+    native: false,
+    md5: <<65, ..., 206>>
+  ]
+  ```
+
+  * Run the ABCI Server's `start_listener` and `stop_listener` functions
```
Comment (KrzysiekJ) [inline README.md:72]: This is already written in `doc/overview.edoc` (though for Erlang).

File: README.md [hunk 4]
```diff
@@ -2,10 +2,87 @@
...
+    ```
+    iex> {ok, _} = :abci_server.start_listener(Foo, 46658)
+    {:ok, #PID<0.181.0>}
+
+    iex> ok = :abci_server.stop_listener(Foo)             
+    :ok
+    ```
+
+  * Test the Running ABCI Server (Erlang) in separate Bash Terminal Tab - https://github.com/tendermint/abci#tools
```
Comment (KrzysiekJ) [inline README.md:81]: This is Tendermint-related documentation, no need to duplicate it here.

File: README.md [hunk 5]
```diff
@@ -2,10 +2,87 @@
...
+  ```
+
+* Install Mix Dependencies
+  ```bash
+  mix deps.get
+  ```
+
+* Documentation Generation. Open Documentation in Web Browser
+  ```bash
+  cd deps/abci_server/ && make docs && open doc/index.html && cd ../../
```
Comment (KrzysiekJ) [inline README.md:30]: This is already covered in the “Documentation” section.

**Comment by ltfschoen** [comment]
Based on feedback from @KrzysiekJ in https://github.com/KrzysiekJ/abci_server/issues/4 I've modified this Pull Request https://github.com/KrzysiekJ/abci_server/pull/3/files and removed the proposed change to the abci_server's Makefile so it still uses Ranch 1.3.2. The Pull Request now only proposes to modify the Readme and includes a Troubleshooting section that may help other users who encounter the IEx error `(Mix) Could not start application ranch: could not find application file: ranch.app` by suggesting that they try adding Ranch 1.4.0 to their Mix dependencies.

File: README.md [hunk 6]
```diff
@@ -2,10 +2,103 @@
...
+    [
+      # ABCI Server (Erlang) - https://github.com/KrzysiekJ/abci_server
+      {:abci_server, git: "https://github.com/KrzysiekJ/abci_server.git", tag: "v0.4.0"}
+    ]
+  end
+  ```
+
+* Install Mix Dependencies
+  ```bash
+  mix deps.get
```
Comment (KrzysiekJ) [inline README.md:25]: This looks like a generic Elixir command, no need to put it here.

**Comment by KrzysiekJ** [comment]
Left some comments. Generally I have a feeling that the docs provided are much redundant, not project-specific. If they get fixed (the current example will not work), they may become valuable as a blog post or tutorial or on wiki. For example, adding Erlang packages as dependencies to Elixir projects (and vice versa) is a generic topic and it should be covered perhaps in Elixir documentation.

**Pull request closed without merging**